test/distributed/fsdp/test_fsdp_summon_full_params.py

```diff
@@ -249,7 +263,9 @@ def test_reshard_outside_forward_backward_iteration(self):
         output = model(torch.zeros(5).cuda(self.rank))
         with model._summon_full_params():
             pass
-        self.assertEqual(outer_full_param_size, outer_param._full_param_padded.storage().size())
+        self.assertEqual(
```

Comment:
Couldn't see if this was already added, but can we add a test that the parameter names are fixed / back to the original module's with this unflattening fix?

Reply:
I added a test to check the number of `fsdp_model.parameters()` and the value of the parameters. Testing keys is more tricky as this relies on `state_dict()`  or `named_parameter()` which both do not call `_summon_full_parameters()` in the current code base.

Reply:
I see, I filed an issue to track this to make sure that we fix it: https://github.com/pytorch/pytorch/issues/72716